### PR TITLE
Adding first UI for create post feature

### PR DESCRIPTION
**Pull request by roncoderpush**
## WHAT
- Adding first UI for create post feature. API interaction and button action are not implemented.

##SCREENSHOT
Simulator Screenshot - iPhone 14 Pro Max - 2023-08-08 at 16 49 01

### CODE REVIEW

File: lib/core/presentation/pages/create_post/create_post_page.dart [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+class CreatePostPage extends StatelessWidget {
+  const CreatePostPage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final textCtrl = TextEditingController();
+    final t = Translations.of(context);
+    return Scaffold(
+      backgroundColor: LemonColor.black,
+      appBar: AppBar(
```
Comment (mthinh) [inline lib/core/presentation/pages/create_post/create_post_page.dart:19]: Please check on using LemonAppBar, already designed-like back button

File: lib/core/presentation/pages/create_post/create_post_page.dart [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            ),
+          ),
+          SizedBox(width: Spacing.smMedium),
+        ],
+        elevation: 0,
+      ),
+      body: SafeArea(
+        child: Column(
+          children: [
+            Divider(color: LemonColor.white.withOpacity(0.7)),
```
Comment (mthinh) [inline lib/core/presentation/pages/create_post/create_post_page.dart:53]: use colorScheme.onSurface

Screenshot 2023-08-09 at 08 24 09

File: lib/core/presentation/pages/create_post/create_post_page.dart [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            Expanded(
+              child: Padding(
+                padding: EdgeInsets.symmetric(horizontal: Spacing.smMedium),
+                child: TextFormField(
+                  controller: textCtrl,
+                  maxLines: 10,
+                  cursorColor: LemonColor.white,
```
Comment (mthinh) [inline lib/core/presentation/pages/create_post/create_post_page.dart:60]: Use color scheme

File: lib/core/presentation/pages/create_post/create_post_page.dart [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                  decoration: InputDecoration.collapsed(hintText: t.home.whatOnYourMind),
+                ),
+              ),
+            ),
+            Divider(color: LemonColor.white.withOpacity(0.7)),
```
Comment (mthinh) [inline lib/core/presentation/pages/create_post/create_post_page.dart:65]: Use color scheme

File: lib/core/presentation/pages/create_post/create_post_page.dart [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            Padding(
+              padding: EdgeInsets.symmetric(horizontal: Spacing.smMedium),
+              child: Row(
+                children: [
+                  Assets.icons.icHouseParty.svg(color: LemonColor.white.withOpacity(0.7)),
```
Comment (mthinh) [inline lib/core/presentation/pages/create_post/create_post_page.dart:70]: `color` property for svg is deprecated. 
may use `ThemeSvgIcon(color: colorScheme.someColor, builder: (filter) => svg(colorFilter: filter)))`

File: lib/core/presentation/pages/create_post/create_post_page.dart [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+  @override
+  Widget build(BuildContext context) {
+    final textCtrl = TextEditingController();
+    final t = Translations.of(context);
+    return Scaffold(
+      backgroundColor: LemonColor.black,
+      appBar: AppBar(
+        automaticallyImplyLeading: true,
+        actions: [
+          InkWell(
```
Comment (mthinh) [inline lib/core/presentation/pages/create_post/create_post_page.dart:22]: can consider using LinearGradientButton as the future design may move to use lavender gradient  button as a main button

File: lib/core/presentation/pages/create_post/create_post_page.dart [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+                  padding: EdgeInsets.symmetric(
+                    horizontal: Spacing.smMedium,
+                    vertical: Spacing.superExtraSmall,
+                  ),
+                  decoration: BoxDecoration(
+                    borderRadius: BorderRadius.circular(32),
+                    color: LemonColor.lavender,
+                  ),
+                  child: Text(
+                    'Post',
```
Comment (mthinh) [inline lib/core/presentation/pages/create_post/create_post_page.dart:36]: define i18n in `post_en.i18n.json`

File: lib/core/presentation/pages/create_post/create_post_page.dart [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+                    style: TextStyle(
+                      fontSize: 14,
```
Comment (mthinh) [inline lib/core/presentation/pages/create_post/create_post_page.dart:38]: use Typo.medium.copyWith(fontWeight: FontWeight.w700)

**Review by mthinh — CHANGES_REQUESTED** [review]
- Color should be use base on colorScheme 
- Font size should be use base on Typo
- Some widgets already available 
- For creating new page, the outer folder should be a domain name like `post`, 
if page is simple, can consider create a dart file with suffix page ex: `post_page.dart`, 
if page is complicated, and may contain domain specific view or widgets, consider create a folder for page `create_post` 
- For commit, please follow conventional commit https://www.conventionalcommits.org/en/v1.0.0/

**Comment by RonTran636** [comment]
@mthinh changes applied, please review again

File: lib/core/presentation/pages/create_post/create_post_page.dart [hunk 9]
```diff
@@ -50,34 +53,49 @@ class CreatePostPage extends StatelessWidget {
...
+                    flex: 7,
+                    child: Row(
+                      children: [
+                        ThemeSvgIcon(
+                          color: colorScheme.onSurface,
+                          builder: (filter) => Assets.icons.icHouseParty.svg(colorFilter: filter),
+                        ),
+                        SizedBox(width: Spacing.medium),
+
+                        ///Temporary remove since we don't have integration with it yet
```
Comment (mthinh) [inline lib/core/presentation/pages/create_post/create_post_page.dart:81]: Please add prefix TODO

File: lib/core/presentation/pages/create_post/create_post_page.dart [hunk 10]
```diff
@@ -50,34 +53,49 @@ class CreatePostPage extends StatelessWidget {
...
+                          builder: (filter) => Assets.icons.icCamera.svg(colorFilter: filter),
+                        ),
+                      ],
+                    ),
+                  ),
+                  Expanded(
+                    flex: 3,
                     child: Container(
                       decoration: BoxDecoration(
                         color: LemonColor.black,
```
Comment (mthinh) [inline lib/core/presentation/pages/create_post/create_post_page.dart:99]: Please use LemonOutlineButton, and don't set fixed color, may affect theming in the future.

File: lib/core/presentation/pages/create_post/create_post_page.dart [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+                  Assets.icons.icCrystal.svg(color: LemonColor.white.withOpacity(0.7)),
+                  SizedBox(width: Spacing.medium),
+                  Assets.icons.icTicket.svg(color: LemonColor.white.withOpacity(0.7)),
+                  SizedBox(width: Spacing.medium),
+                  Assets.icons.icPoll.svg(color: LemonColor.white.withOpacity(0.7)),
+                  SizedBox(width: Spacing.medium),
+                  Assets.icons.icCamera.svg(color: LemonColor.white.withOpacity(0.7)),
+                  SizedBox(width: Spacing.medium),
+                  Expanded(
+                    child: Container(
```
[the remark this answers is not in the document]
Comment (mthinh) [reply lib/core/presentation/pages/create_post/create_post_page.dart:97]: @roncoderpush can you check on this ?

File: lib/core/presentation/pages/create_post/create_post_page.dart (hunk 6 above)
Comment (mthinh) [reply lib/core/presentation/pages/create_post/create_post_page.dart:26]: @roncoderpush can you check on this pls ?

**Review by mthinh — CHANGES_REQUESTED** [review]
Some changes needed @roncoderpush

**Comment by roncoderpush** [comment]
@mthinh changes applied, please review again

File: lib/core/presentation/pages/create_post/create_post_page.dart (hunk 1 above)
Comment (RonTran636) [reply lib/core/presentation/pages/create_post/create_post_page.dart:24]: LemonAppBar are using specified space for icons only. We having button here so cannot re-use it

**Comment by RonTran636** [comment]
@mthinh  I change 2 requests for the button. The outline button color doesn't match the UI. Please check.
Simulator Screenshot - iPhone 14 Pro Max - 2023-08-11 at 03 07 31

File: lib/core/presentation/pages/create_post/create_post_page.dart (hunk 1 above)
Comment (mthinh) [reply lib/core/presentation/pages/create_post/create_post_page.dart:24]: As I remember it allows to pass actions as a list of widgets  :/

**Comment by mthinh** [comment]
> @mthinh I change 2 requests for the button. The outline button color doesn't match the UI. Please check. Simulator Screenshot - iPhone 14 Pro Max - 2023-08-11 at 03 07 31

@roncoderpush you mean which color ? the border color in the design is White.opacity(0.12) which is colorScheme.outline, The outline button currently uses that color
Screenshot 2023-08-11 at 08 37 43

**Pull request merged**